File: app/src/main/java/org/zeroxlab/momodict/widget/PagerFocusBroadcaster.java
package org.zeroxlab.momodict.widget;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;

import org.zeroxlab.momodict.MainActivity;

public class PagerFocusBroadcaster implements ViewPager.OnPageChangeListener {

    private final MainActivity.FragmentPagerAdapterImpl mAdapter;

    public PagerFocusBroadcaster(@NonNull MainActivity.FragmentPagerAdapterImpl adapter) {
        mAdapter = adapter;
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
    }

    @Override
    public void onPageSelected(int position) {
        notifyViewPager(position);
    }

    @Override
    public void onPageScrollStateChanged(int state) {
    }

    private void notifyViewPager(int pos) {
        Fragment item = mAdapter.getFragment(pos);
        if ((item != null) && (item instanceof ViewPagerFocusable)) {
            ((ViewPagerFocusable) item).onViewPagerFocused();
        }
    }
}

File: app/src/main/java/org/zeroxlab/momodict/widget/PagerFocusBroadcaster.kt
package org.zeroxlab.momodict.widget

import androidx.viewpager.widget.ViewPager
import org.zeroxlab.momodict.MainActivity

class PagerFocusBroadcaster(private val mAdapter: MainActivity.FragmentPagerAdapterImpl) : androidx.viewpager.widget.ViewPager.OnPageChangeListener {

    override fun onPageSelected(position: Int) = notifyViewPager(position)

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}

    override fun onPageScrollStateChanged(state: Int) {}

    private fun notifyViewPager(pos: Int) {
        mAdapter.getFragment(pos)?.let { frg ->
            if (frg is ViewPagerFocusable) {
                frg.onViewPagerFocused()
            }
        }
    }
}
